fil_fungible_token/src/token/mod.rs

```diff
@@ -1043,8 +1071,9 @@ mod test {
         };
 
         // state remained unchanged
-        assert_eq!(token.balance_of(TREASURY).unwrap(), TokenAmount::zero());
-        assert_eq!(token.total_supply(), TokenAmount::zero());
+        // TODO: we abort rather than manually revert state, probably need to get rid of these checks
+        //assert_eq!(token.balance_of(TREASURY).unwrap(), TokenAmount::zero());
+        //assert_eq!(token.total_supply(), TokenAmount::zero());
```

Comment:
Should this just be removed now that the state isn't reverted when receiver hook aborts? I think checking this post-failure state now needs to be done in an integration test

Reply:
Will the library give any utilities to assist the state revert? If so, it can be tested here ("mocking" the expected call). 

Else, I think it can only be tested at an actor level

Reply:
Added a `replace` function to `Token` to do it, so we can clone the original state and swap over to that in place of the modified state where tests need us to revert. Trying to use it outside of tests would be a swift trip to borrow-checker hell so it's limited by config and visibility flags to tests only.